controllers/salons.js

```diff
@@ -0,0 +1,176 @@
+const Moment = require('moment');
+const MomentRange = require('moment-range');
+ 
+const moment = MomentRange.extendMoment(Moment);
+
+const Salon = require('../models/salon')
+
+exports.freeHours = async (req, res, next) => {
+    try {
+        const salonId = req.body.salonId
+        const serviceId = req.body.serviceId
+        const rangeStart = moment(req.body.timeRange.start).utc()
+        const rangeEnd = moment(req.body.timeRange.end).utc()
+
+        const salon = await Salon.findOne({_id: salonId})
+        if(!salon) {
+            const error = new Error('selected salon not exists')
+            error.statusCode = 404
+            throw error
+        }
+
+        const [ startDayOfWeek ] = salon.openingHours.filter(day => {
+            return day.name.toLowerCase() === rangeStart.format('dddd').toLowerCase()
+        })
+
+        const [ endDayOfWeek ] = salon.openingHours.filter(day => {
+            return day.name.toLowerCase() === rangeEnd.format('dddd').toLowerCase()
+        })
+
+        if(!startDayOfWeek || !endDayOfWeek) {
+            const error = new Error('salon is close in selected day')
+            error.statusCode = 500
+            throw error
+        }
+
+        if((rangeStart.format("hh:mm") < startDayOfWeek.open || rangeStart.format("hh:mm") > startDayOfWeek.close) || (rangeEnd.format("hh:mm") > endDayOfWeek.close || rangeEnd.format("hh:mm") < endDayOfWeek.open)) {
+            const error = new Error('salon is close in selected range')
+            error.statusCode = 500
+            throw error
+        }
+
+        const [ service ] = salon.services.filter(service => {
+            return service._id.toString() === serviceId
+        })
+
+        if(!service) {
+            const error = new Error('selected service not exists')
+            error.statusCode = 500
```

Comment:
I think in every case status code 500 isn't the right choice. It should be changed.